stdlib/prettyprint/0/prettyprint.rbs

```diff
@@ -0,0 +1,364 @@
+# frozen_string_literal: true
+#
+# This class implements a pretty printing algorithm. It finds line breaks and
+# nice indentations for grouped structure.
+#
+# By default, the class assumes that primitive elements are strings and each
+# byte in the strings have single column in width. But it can be used for other
+# situations by giving suitable arguments for some methods:
+# *   newline object and space generation block for PrettyPrint.new
+# *   optional width argument for PrettyPrint#text
+# *   PrettyPrint#breakable
+#
+#
+# There are several candidate uses:
+# *   text formatting using proportional fonts
+# *   multibyte characters which has columns different to number of bytes
+# *   non-string formatting
+#
+#
+# ## Bugs
+# *   Box based formatting?
+# *   Other (better) model/algorithm?
+#
+#
+# Report any bugs at http://bugs.ruby-lang.org
+#
+# ## References
+# Christian Lindig, Strictly Pretty, March 2000,
+# http://www.st.cs.uni-sb.de/~lindig/papers/#pretty
+#
+# Philip Wadler, A prettier printer, March 1998,
+# http://homepages.inf.ed.ac.uk/wadler/topics/language-design.html#prettier
+#
+# ## Author
+# Tanaka Akira <akr@fsij.org>
+class PrettyPrint
+  # This is a convenience method which is same as follows:
+  #
+  #     begin
+  #       q = PrettyPrint.new(output, maxwidth, newline, &genspace)
+  #       ...
+  #       q.flush
+  #       output
+  #     end
+  #
+  def self.format: (?untyped output, ?Integer maxwidth, ?String newline, ?untyped genspace) ?{ (Integer) -> Integer } -> untyped
```

Comment:
I think we can make an interface with `<<` method, and use the interface as the type of `output` here?

Reply:
If I correctly understood something like: 

```
interface  _Output
  def <<: (?String obj, ?String sep, ?String newline, ?untyped genspace ) ?{ (Integer) -> Integer } -> void
end
```
An then the definition becomes

```
   def self.format: (?untyped output, ?Integer maxwidth, ?String newline, ?untyped genspace) ?{ (Integer) -> Integer } -> Output
```

Reply:
I mean like the following:

```rbs
class PrettyPrint
  interface _Output
    def <<: (String) -> void
  end

  def self.format: (?_Output output, ?Integer maxwidth, ?String newline, ?untyped genspace) ?{ (Integer) -> Integer } -> _Output # !?!?
end
```

Yes, it's my bad... I finally found that the type cannot be written in RBS... 😢 

Leave the type of the first argument `?untyped`. 🙏 (I'm sorry for the confusion.)

Reply:
No worries at all :)